test/asyncworker.cc

```diff
@@ -9,14 +9,25 @@ class TestWorker : public AsyncWorker {
     Object resource = info[1].As<Object>();
     Function cb = info[2].As<Function>();
     Value data = info[3];
+    bool setResult = info[4].As<Boolean>();
 
     TestWorker* worker = new TestWorker(cb, "TestResource", resource);
     worker->Receiver().Set("data", data);
     worker->_succeed = succeed;
+    worker->_setResult = setResult;
     worker->Queue();
   }
 
 protected:
+  std::vector<napi_value> GetResult(Napi::Env env) override {
```

Comment:
I think it would be good to keep some tests that validate the default case.  Maybe we can add TestWorkerNoReturnValues and use that in 1 or more tests to validate the case were setResult is not overridden.

Reply:
So I reverted back to the old test cases, and added an `AsyncWorkerWithResult` that does override the `GetResult()` method. I think this suffices your concern?

Reply:
Perfect.